src/main/components/canvas/mouse-eventlistener.tsx

```diff
@@ -0,0 +1,273 @@
+    // we want to continue with the selection box from where we left off
+    if (this.state.selectionStarted) {
+      this.setState((prevState) => {
+        return {
+          ...prevState,
+          selectionRectangle: {
+            ...prevState.selectionRectangle,
+            endX: event.clientX,
+            endY: event.clientY,
+          },
+        };
+      });
+
+      return;
+    }
+
+    // the selection box will activate when clicking anywhere outside the bounds of an element
+    // however:
+    //      * resizing an element can start when clicking slightly outside its bounds
+    //      * the connection/reconnection port of an element is outside its bounding box
+    // in these cases the selection box needs to be disabled
+    if (
+      this.props.resizingInProgress ||
+      this.props.connectingInProgress ||
+      this.props.reconnectingInProgress ||
+      this.props.hoveringInProgress
+    ) {
+      return;
+    }
+
+    this.props.changeSelectionBox(true);
+
+    this.setState({
+      selectionStarted: true,
+      selectionRectangle: {
+        startX: event.clientX,
+        startY: event.clientY,
+        endX: undefined,
+        endY: undefined,
+      },
+    });
+  };
+
+  /**
+   * Mouse up handler for finalising the box selection and determining which elements to select
+   */
+  private mouseUp = (): void => {
+    // if no selection has been started, we can skip determining which
+    // elements are contained in the selection box.
+    if (!this.state.selectionStarted) {
+      return;
+    }
+
+    const selection = this.getElementIDsInSelectionBox();
+    this.props.select(selection);
+
+    this.setState({
+      selectionStarted: false,
+      selectionRectangle: {
+        startX: undefined,
+        startY: undefined,
+        endX: undefined,
+        endY: undefined,
+      },
+    });
+
+    this.props.changeSelectionBox(false);
+  };
+
+  /**
+   * Mouse move handler for dragging the selection rectangle
+   * @param event The triggering mouse move event
+   */
+  private mouseMove = (event: MouseEvent): void => {
+    if (!this.state.selectionStarted) {
+      return;
+    }
+
+    const elementsInSelectionBox = this.getElementIDsInSelectionBox();
+
+    this.setState((prevState) => {
+      return {
+        selectionStarted: prevState.selectionStarted,
+        elementsInSelectionBox,
+        selectionRectangle: {
+          ...prevState.selectionRectangle,
+          endX: event.clientX,
+          endY: event.clientY,
+        },
+      };
+    });
+  };
+
+  /**
+   * Check whether a given IUMLElement is contained in the currently active selection rectangle.
+   * Elements are only considered selected if they are fully contained within the selection rectangle.
+   *
+   * @param element The element for which containment in the selection box is determined
+   */
+  private isElementInSelectionBox = (element: IUMLElement): boolean => {
```

Comment:
I suspect this method should be a utility function in geometry, i.e. perhaps [here](https://github.com/ls1intum/Apollon/blob/develop/src/main/utils/geometry/boundary.ts).

Reply:
Additionally, this right now only selects an element if it is fully inside the selection box, while I suspect the more user-friendly way would be to select an element if its boundary overlaps with the selection box enough.